src/Network/MQTT/Encoder.hs

```diff
@@ -0,0 +1,166 @@
+{-# LANGUAGE OverloadedStrings, RecordWildCards #-}
+module Network.MQTT.Encoder (encodePacket) where
+
+import           Data.Bits (shiftL, (.|.))
+import           Data.ByteString.Builder
```

Comment:
Specify import list. In general, you should specify import list for all non-qualified imports.